awx/conf/settings.py

```diff
@@ -437,11 +447,15 @@ def SETTINGS_MODULE(self):
         return self._get_default('SETTINGS_MODULE')
 
     def __getattr__(self, name):
+        if name.startswith('LOG_AGGREGATOR_'):
```

Comment:
cc @AlanCoding I think this is going to help *a lot* with performance across the board when external logging in enabled

Reply:
> I think this is going to help a lot with performance across the board when external logging is enabled

It will help when external logging is _not_ enabled. When we replaced service restarts with more frequent hits to memcache, this included tests for _enablement_.

I'm wondering if there is anything we need to document this. And I think the answer is yes, but not much.

https://github.com/ansible/awx/blob/c33d2a1e00abb137047d96d17d8763a8e0e7265d/awx/main/conf.py#L701-L709

Within this setting help text, I would suggest

> Enable sending logs to external log aggregator. This will take a minimum of 10 seconds to take effect, as will any other change to logging settings.

I think that this per-process TTL _could_ be rolled out as a parameter to settings definitions generally. In most cases, it probably still makes sense to not cache it locally like this, but there are still many cases where we check a value multiple times within the same request / operation, which is never a thing we should want to do.

Reply:
That's true, but it yields even more of a boost when logging is enabled.

I've set it to 5 seconds, so I question whether we need to do much to actually tell people about this, especially since the old behavior was that we restarted everything.  I think it's reasonable that it takes a few seconds for logs to show up.

Reply:
cc @gamuniz